tests/shell/mod.rs

```diff
@@ -231,61 +231,61 @@ fn run_export_extern() {
 #[test]
 #[cfg(not(windows))]
 fn run_in_login_mode() {
-    let child_output = std::process::Command::new("sh")
+    let status = std::process::Command::new("sh")
         .arg("-c")
         .arg(format!(
-            "{:?} -l -c 'echo $nu.is-login'",
+            "{:?} --no-config-file --login --commands 'echo $nu.is-login'",
             nu_test_support::fs::executable_path()
         ))
-        .output()
+        .status()
```

Comment:
Do we always correctly set the status code for this to be sensitive enough?

Reply:
I hope so, otherwise I would consider that a bug 😅

Reply:
I don't have that faith for the tests. They should stop us from those bugs 😅

Reply:
I had a quick check and I couldn't find an obvious way where it would print to stderr but not return 1 as exit code.
Also printing to stderr doesn't always mean there's an error, right?

But I am happy to revert these changes if you want me to.

Reply:
Also true. 

I think we have a tendency to rely on the stderr for a lot of our failure tests, so I am not sure how well exercised the statuscode path is.

If that is fine for those tests here I am ok with this as it reads simpler.